packages/salesforcedx-vscode-core/src/commands/forceStopApexDebugLogging.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  CommandOutput,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as vscode from 'vscode';
+import { developerLogTraceFlag } from '.';
+import { nls } from '../messages';
+import { hideTraceFlagExpiration } from '../traceflag-time-decorator';
+import {
+  CancelResponse,
+  ContinueResponse,
+  ParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+class ForceStopApexDebugLoggingExecutor extends SfdxCommandletExecutor<{}> {
+  public build(data: {}): Command {
+    return getRestoreLevelsCommand();
+  }
+
+  public execute(response: ContinueResponse<{}>): void {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: vscode.workspace.rootPath
+    }).execute(cancellationToken);
+
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+    execution.processExitSubject.subscribe(async data => {
+      if (data != undefined && data.toString() === '0') {
+        developerLogTraceFlag.turnOffLogging();
+        hideTraceFlagExpiration();
+      }
+    });
+  }
+}
+
+export async function restoreDebugLevels(): Promise<void> {
+  if (developerLogTraceFlag.isActive()) {
+    const execution = new CliCommandExecutor(getRestoreLevelsCommand(), {
+      cwd: vscode.workspace.rootPath
+    }).execute();
+    const resultPromise = new CommandOutput().getCmdResult(execution);
+    const result = await resultPromise;
+    const resultJson = JSON.parse(result);
+    if (resultJson.status === 0) {
+      return Promise.resolve();
+    } else {
+      return Promise.reject('Restoring the debug levels failed.');
```

Comment:
If this message is user facing, add this to i18n.ts and you may need to suggest what the user can do if this fails. cc @ruthemmanuelle

Reply:
It does not look like this is user facing - I looked through the source and they do not do anything if the promise rejects and its not the shutdown case. If it is the shut down case, then it should not show since the extension host is shut down.

https://github.com/Microsoft/vscode/blob/3dd2035b13b3395f55afd5070ba875bcc230e464/src/vs/workbench/api/node/extHostExtensionService.ts#L271